osu.Game/Screens/Select/FooterV2/BeatmapOptionsPopover.cs

```diff
@@ -0,0 +1,148 @@
+using osu.Game.Graphics.UserInterfaceV2;
+using osu.Game.Overlays;
+using osuTK;
+using osuTK.Graphics;
+using osuTK.Input;
+
+namespace osu.Game.Screens.Select.FooterV2
+{
+    public partial class BeatmapOptionsPopover : OsuPopover
+    {
+        private FillFlowContainer<OptionButton> buttonFlow = null!;
+        private readonly FooterButtonOptionsV2 footerButton;
+
+        public BeatmapOptionsPopover(FooterButtonOptionsV2 footerButton)
+        {
+            this.footerButton = footerButton;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(ManageCollectionsDialog? manageCollectionsDialog, SongSelect? songSelect, OsuColour colours)
+        {
+            Content.Padding = new MarginPadding(5);
+
+            Child = buttonFlow = new FillFlowContainer<OptionButton>
+            {
+                AutoSizeAxes = Axes.Both,
+                Direction = FillDirection.Vertical,
+                Spacing = new Vector2(3),
+            };
+
+            addButton(@"Manage collections", FontAwesome.Solid.Book, () => manageCollectionsDialog?.Show());
+            addButton(@"Delete all difficulties", FontAwesome.Solid.Trash, () => songSelect?.DeleteBeatmap(), colours.Red);
+            addButton(@"Remove from unplayed", FontAwesome.Regular.TimesCircle, null);
+            addButton(@"Clear local scores", FontAwesome.Solid.Eraser, () => songSelect?.ClearScores());
+
+            if (songSelect != null && songSelect.AllowEditing)
+                addButton(@"Edit beatmap", FontAwesome.Solid.PencilAlt, () => songSelect.Edit());
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            ScheduleAfterChildren(() => GetContainingInputManager().ChangeFocus(this));
+        }
+
+        private void addButton(LocalisableString text, IconUsage icon, Action? action, Color4? colour = null)
+        {
+            var button = new OptionButton
+            {
+                Text = text,
+                Icon = icon,
+                TextColour = colour,
+                Action = () =>
+                {
+                    Hide();
+                    action?.Invoke();
+                },
+            };
+
+            buttonFlow.Add(button);
+        }
+
+        private partial class OptionButton : OsuButton
+        {
+            public IconUsage Icon { get; init; }
+            public Color4? TextColour { get; init; }
+
+            public OptionButton()
+            {
+                Size = new Vector2(265, 50);
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OverlayColourProvider colourProvider)
+            {
+                BackgroundColour = colourProvider.Background3;
+
+                SpriteText.Colour = TextColour ?? Color4.White;
+                Content.CornerRadius = 10;
+
+                Add(new SpriteIcon
+                {
+                    Blending = BlendingParameters.Additive,
```

Comment:
Why though? This makes the icon appear as if it has a completely different colour than the text. Doesn't seem good...

Reply:
It was copied here:
https://github.com/ppy/osu/blob/a200d043be9b717da0d72062fe7a5466372ef5c8/osu.Game/Screens/Edit/Components/EditorToolButton.cs#L57

Will remove.